### PR TITLE
changed paramter "index" to "axis" for signature matching with numpy

### PR DESCRIPTION
**Pull request by jmcq**
changed "stdev()" to "std()", "avg()" to "average()" for signature match with numpy

### CODE REVIEW

**Comment by ChrisBeaumont** [comment]
I agree that changing `index` to `axis` would make more sense to most users, since these functions are meant to mimic  the numpy API anyways (which uses `axis`). My only hestitation here is whether we should make this change in one step, or support both keywords for 1 release, emitting deprecation warnings if a user uses `index`.

**Comment by jmcq** [comment]
That's up to you and Bryan to tell me how many people are using this.  I
can't afford to support things that no one is actually using.
So far as I know, I am supporting our first commercial customer of the
Python API (but I could be wrong).

The other possibility is to make a fork and play out an API that sticks to
numpy.  But again, I don't know what is the right strategy ... I am a
submitter at this point, but a long-time numpy user.

Bryan?

**Comment by jmcq** [comment]
Don't know github well enough to know if Bryan gets a response to that.
Adding him manually.

Bryan please see my comment to Chris below.

**Comment by bwlewis** [comment]
Based on the high interest level of this project, I think Chris is right to take a conservative opinion and just support both function names for a while, rather than make a breaking change.

**Comment by ChrisBeaumont** [comment]
@jmcq I think this is a good change, and it wouldn't be hard to support both syntaxes for a while. We should be able to define a decorator along these lines (caveat: untested code):

```python
from functools import wraps
import warnings

def deprecated_index(func):

    @wraps(func)
    def wrapper(self, *args, **kwargs):
        if 'index' in kwargs:
            warnings.warn("`index` is deprecated, and will be removed in the next version of SciDB-Py. Use `axis` instead", DeprecationWarning)
            kwargs['axis'] = kwargs.pop('index')
        return func(self, *args, **kwargs)
    
    return wrapper
```

and then add `@deprecated_index` above each of the functions that have changed. Is that something you'd be able to add to this PR?

**Comment by ChrisBeaumont** [comment]
Also, just for clarification, the `scidb_syntax` keywords in these functions is meant to allow users to switch between SciDB's convention for aggregations (collapse over all dimensions *except* this dimension) and numpy's (collapse over *only* this dimension). That's definitely a distinction we need to retain, since both usages are common. The naming of axis vs index is much less important to preserve after a deprecation period.

**Comment by jmcq** [comment]
Sure, if there are lots of people using the library then its worth it.  But
you didn't answer my question -- how many people are using it.

"high interest level of this project" isn't meaningful to me, possibly
because I don't know how to use GitHub yet.  Can you tell me what that
means in # of downloads, # of SciDB paying customers, or something like
that?

**Comment by ChrisBeaumont** [comment]
> Sure, if there are lots of people using the library then its worth it. 

I don't know how many people use SciDB-Py (either total or as a fraction of SciDB users), but I know from forum posts and emails that it's >0, and I'm personally not comfortable with breaking the API without going through the standard deprecation process.

File: scidbpy/interface.py [hunk 1]
```diff
@@ -1057,74 +1057,82 @@ def isnan(self, A):
         """Element-wise nan test function"""
         return self._apply_func(A, 'is_nan')
 
-    def min(self, A, index=None, scidb_syntax=False):
+    def min(self, A, axis=None, scidb_syntax=False):
         """
         Array or axis minimum.
 
         see :meth:`SciDBArray.min` """
         return A.min(index, scidb_syntax)
```
Comment (ChrisBeaumont) [inline scidbpy/interface.py]: Note that you also need to change `index` to `axis` in the bodies of these functions

File: scidbpy/interface.py [hunk 2]
```diff
@@ -1057,74 +1057,82 @@ def isnan(self, A):
...
 
-    def approxdc(self, A, index=None, scidb_syntax=False):
+    def approxdc(self, A, axis=None, scidb_syntax=False):
         """
         Array or axis unique element estimate.
 
-        see :meth:`SciDBArray.approxdc` """
+        see :meth:`SciDBArray.approxdc`
+
+        NOTE: not numpy-like.  may become .afl.approxdc
```
Comment (ChrisBeaumont) [inline scidbpy/interface.py]: We don't need to add caveats for non-numpy methods unless numpy also has a method with the same name that behaves differently

**Pull request closed without merging**